src/ssl_cert_paths_discover.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2010-2021, Tarantool AUTHORS, please see AUTHORS file.
+ *
+ * Redistribution and use in source and binary forms, with or
+ * without modification, are permitted provided that the following
+ * conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above
+ *    copyright notice, this list of conditions and the
+ *    following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above
+ *    copyright notice, this list of conditions and the following
+ *    disclaimer in the documentation and/or other materials
+ *    provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY <COPYRIGHT HOLDER> ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * <COPYRIGHT HOLDER> OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+ * BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <dirent.h>
+#include <sys/stat.h>
+#include <openssl/ssl.h>
+#include "ssl_cert_paths_discover.h"
+#include "tt_static.h"
+
+static int
+is_dir_empty(const char *dir_path)
+{
+	DIR *dir = opendir(dir_path);
+	if (!dir)
+		return 1;
+
+	struct dirent *ent;
+	int is_empty = 1;
+	while ((ent = readdir(dir))) {
+		if (strcmp(ent->d_name, ".") && strcmp(ent->d_name, "..")) {
+			if (ent->d_type == DT_REG ||
+			    ent->d_type == DT_LNK ||
+			    ent->d_type == DT_DIR) {
+				is_empty = 0;
+				break;
+			}
+		}
+	}
+
+	closedir(dir);
+	return is_empty;
+}
+
+const char *DEFAULT_CERT_DIR_PATHS[] = {
+	/* Fedora/RHEL/Centos */
+	"/etc/pki/tls/certs",
+	/* Debian/Ubuntu/Gentoo etc. (OpenSuse) */
+	"/etc/ssl/certs",
+	/* FreeBSD */
+	"/usr/local/share/certs",
+	/* NetBSD */
+	"/etc/openssl/certs",
+	/* macOS */
+	"/private/etc/ssl/certs",
+	"/usr/local/etc/openssl@1.1/certs",
+	"/usr/local/etc/openssl@1.0/certs",
+	"/usr/local/etc/openssl/certs",
+	NULL
+};
+
+const char *DEFAULT_CERT_FILE_PATHS[] = {
+	/* Fedora/RHEL 6 */
+	"/etc/pki/tls/certs/ca-bundle.crt",
+	/* CentOS/RHEL 7 */
```

Comment:
What's about CentOS/RHEL 8?

Reply:
The same path as for Centos7. I've fixed comment